systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java

```diff
@@ -269,6 +266,73 @@ void testKafkaAndZookeeperScaleUpScaleDown() throws Exception {
         receiveMessagesExternal(NAMESPACE, topicName, messageCount);
     }
 
+    /**
+     * This test cover case, when KafkaRoller will not roll Kafka pods, because created topic doesn't meet requirements for roll remaining pods
+     * 1. Deploy kafka cluster with 3 pods
+     * 2. Create topic with 4 replicas
+     * 3. Scale kafka cluster to 7 replicas and wait, until all pods are ready
+     * 4. Scale down kafka cluster to 3 replicas
+     * 5. Trigger rolling update for Kafka cluster
+     * 6. Rolling update will not be performed, because topic which we created had some replicas on deleted pods - manual fix is needed in that case
+     */
+    @Test
+    void testKafkaWontRollUp() throws Exception {
+        String topicName = "test-topic-" + new Random().nextInt(Integer.MAX_VALUE);
+        timeMeasuringSystem.setOperationID(timeMeasuringSystem.startTimeMeasuring(Operation.CLUSTER_RECOVERY));
+
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, 3)
+            .editSpec()
+                .editKafka()
+                    .addToConfig("auto.create.topics.enable", "false")
+                .endKafka()
+            .endSpec().done();
+
+        LOGGER.info("Running kafkaScaleUpScaleDown {}", CLUSTER_NAME);
+        final int initialReplicas = kubeClient().getStatefulSet(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).getStatus().getReplicas();
+        assertEquals(3, initialReplicas);
+
+        // scale up
```

Comment:
You don't seem to be doing anything between scale-up and the initial deployment. So why not just deploy the cluster with 4 nodes?

Reply:
Removed scaleUp phase